cobra/core/Model.py

```diff
@@ -46,18 +57,65 @@ def __init__(self, id_or_model=None, name=None):
             self.metabolites = DictList()  # A list of cobra.Metabolites
             # genes based on their ids {Gene.id: Gene}
             self.compartments = {}
-            self.solution = Solution(None)
+            # self.solution = Solution(None)
             self.media_compositions = {}
 
+            # from cameo ...
+
+            # if not hasattr(self, '_solver'):  # backwards compatibility
+            # with older cobrapy pickles?
+            self._solver = solver_interface.Model()
+            self._solver.objective = solver_interface.Objective(S.Zero)
+            self._populate_solver(self.reactions, self.metabolites)
+        self._timestamp_last_optimization = None
+        self.solution = LazySolution(self)
+
+    @property
+    def solver(self):
+        """Get or set the attached solver instance.
+
+        Very useful for accessing the optimization problem directly.
+        Furthermore, can be used to define additional non-metabolic
+        constraints.
+
+        Examples
+        --------
+        >>> import cobra.test
+        >>> model = cobra.test.create_test_model("textbook")
+        >>> new = model.solver.interface.Constraint(model.objective.expression,
+        >>> lb=0.99)
+        >>> model.solver.add(new)
+        """
+        return self._solver
+
+    @solver.setter
+    def solver(self, value):
+        not_valid_interface = ValueError(
+            '%s is not a valid solver interface. Pick from %s, or specify an '
+            'optlang interface (e.g. optlang.glpk_interface).' % (
+                value, list(config.solvers.keys())))
+        if isinstance(value, six.string_types):
+            try:
+                interface = config.solvers[value]
+            except KeyError:
+                raise not_valid_interface
+        elif isinstance(value, types.ModuleType) and hasattr(value, 'Model'):
+            interface = value
+        else:
+            raise not_valid_interface
+        for reaction in self.reactions:
+            reaction._reset_var_cache()
+        self._solver = interface.Model.clone(self._solver)
```

Comment:
That was how solvers were changed previously in cameo. The problem is though, that any additional constraints or variables (e.g. flux ration constraints or auxiliary variables) will be lost by repopulating the solver. Cloning the solver keeps these extra variables and constraints but is a bit slower. Since you would not usually change the solver back and forth a lot, we decided that a few extra seconds were ok to add this functionality.